src/vm/gcenv.ee.cpp

```diff
@@ -1175,7 +1114,204 @@ bool GCToEEInterface::IsGCThread()
+            args->HasStarted = !!args->Thread->HasStarted(false);
+
+            Thread* thread = args->Thread;
+            auto threadStart = args->ThreadStart;
+            void* threadArgument = args->Argument;
+            bool hasStarted = args->HasStarted;
+            args->ThreadStartedEvent.Set();
+
+            // The stubArgs cannot be used once the event is set, since that releases wait on the
+            // event in the function that created this thread and the stubArgs go out of scope.
+            if (hasStarted)
+            {
+                threadStart(threadArgument);
+                DestroyThread(thread);
+            }
+
+            return 0;
+        };
+
+        WCHAR threadName[MaxThreadNameSize];
+        LPWSTR wideName = nullptr;
+        if (MultiByteToWideChar(CP_ACP, 0, name, -1 /* null terminated */, threadName, MaxThreadNameSize))
+        {
+            // use the name given to us if the utf16 conversion was successful.
+            // if not, we're not obligated to give a name to a thread - we'll leave it
+            // as null.
+            wideName = threadName;
+        }
+
+        if (!args.Thread->CreateNewThread(0, threadStub, &args, wideName))
+        {
+            args.Thread->DecExternalCount(FALSE);
+            return false;
+        }
+
+        args.Thread->SetBackground(TRUE, FALSE);
+        args.Thread->StartThread();
+
+        // Wait for the thread to be in its main loop
+        uint32_t res = args.ThreadStartedEvent.Wait(INFINITE, FALSE);
+        args.ThreadStartedEvent.CloseEvent();
+        _ASSERTE(res == WAIT_OBJECT_0);
+
+        if (!args.HasStarted)
+        {
+            // The thread has failed to start and the Thread object was destroyed in the Thread::HasStarted
+            // failure code path.
+            return false;
+        }
+
+        return true;
+    }
+
+    bool CreateUnsuspendableThread(
+        void (*threadStart)(void*),
+        void* argument,
+        const char* name)
+    {
+        LIMITED_METHOD_CONTRACT;
+
+        ThreadStubArguments args;
+        args.Argument = argument;
+        args.ThreadStart = threadStart;
+        args.Thread = INVALID_HANDLE_VALUE;
+        if (!args.ThreadStartedEvent.CreateAutoEventNoThrow(FALSE))
+        {
+            return false;
+        }
+
+        auto threadStub = [](void* argument) -> DWORD
+        {
+            ThreadStubArguments* args = static_cast<ThreadStubArguments*>(argument);
+            assert(args != nullptr);
+
+            ClrFlsSetThreadType(ThreadType_GC);
+            STRESS_LOG_RESERVE_MEM(GC_STRESSLOG_MULTIPLY);
+
+            // We commit the thread's entire stack to ensure we're robust in low memory conditions.
+            BOOL fSuccess = Thread::CommitThreadStack(NULL);
+            if (!fSuccess)
+            {
+#ifdef BACKGROUND_GC
+                // For background GC we revert to doing a blocking GC.
+                return;
```

Comment:
It the `BACKGROUND_GC` defined when compiling VM?

Reply:
(Also, CommitThreadStack is no-op in CoreCLR. You can just delete the whole block.)

Reply:
Probably not - I've been looking around and it looks like this is supposed to be defined by `gcpriv.h`, so it won't have any effect in VM code.

Evidence that it does nothing is this block of code: https://github.com/dotnet/coreclr/blob/master/src/vm/finalizerthread.cpp#L1121-L1126, which *definitely* would not compile if `BACKGROUND_GC` were defined...

at any rate, I'll just delete the block.